shift_detector/precalculations/lda_embedding.py

```diff
@@ -145,21 +151,14 @@ def process(self, store):
                     model = copy(self.model)
                     model = model.fit(vectorized_merged[col])
                     self.trained_model = model
-                transformed1[col] = self.trained_model.transform(vectorized1[col])
-                transformed2[col] = self.trained_model.transform(vectorized2[col])
-
-        for i, col in enumerate(col_names):
-            vec1 = [0] * len(transformed1[col])
-            vec2 = [0] * len(transformed2[col])
-            # infer topics for train_df
-            for j in range(len(transformed1[col])):
-                # take always the topic with the highest probability
-                vec1[j] = transformed1[col][j].argmax()
-            topics1[topic_labels[i]] = vec1
-            # infer topics for test_df
-            for k in range(len(transformed2[col])):
-                # take always the topic with the highest probability
-                vec2[k] = transformed2[col][k].argmax()
-            topics2[topic_labels[i]] = vec2
+
+                # Always takes the topic with the highest probability as the dominant topic
+                transformed1[col] = [arr1.argmax() for arr1 in self.trained_model.transform(vectorized1[col])]
+                transformed2[col] = [arr2.argmax() for arr2 in self.trained_model.transform(vectorized2[col])]
+
+        topics1 = pd.DataFrame(transformed1)
```

Comment:
This part can be removed if you would create `transformed1` and `transformed2` as DataFrames.